runner/config_schema.yaml

```diff
@@ -1,54 +1,126 @@
 $schema: http://json-schema.org/schema#
 $id: http://illixr.github.io/schemas/config.json
-type: object
 definitions:
-  key_vals: &key_vals
-    default: {}
-    type: object
-    additionalProperties:
-      type: string    
+  path:
```

Comment:
This schema is fairly complex now. My concern is whether people will be able to easily follow it and figure out what is going on in the config files.